script.js

```diff
@@ -1,4 +1,198 @@
+//Adding username
+var userName = "";
+
+//Wins and total games played
+var numWins = 0;
+var numTotalGames = 0;
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  //Thanks to Tinaes for teaching array
+  console.log(userName + ": " + input);
+  var moves = ["scissors", "paper", "stone"];
+
+  var compMove = Math.floor(Math.random() * moves.length); //Computer random moves
+
+  var rMoves = ["reversed scissors", "reversed paper", "reversed stone"];
+
+  var rCompMove = Math.floor(Math.random() * rMoves.length); //RComputer random moves
+
+  //Setting username with the main input (reference from solution)
+  if (!userName) {
+    //Need to be first action if not will not execute
+    if (!input) {
+      //What to do if no input yet.
+      return "Please input something as your username!";
+    }
+    userName = input;
+    return (
+      "Welcome Player " +
+      userName +
+      "!<br>Type scissors, paper or stone to play!"
+    );
+  }
+  //Improve with .toLowerCase() to make input case insensitive!
+
+  //If Tie
+  if (input.toLowerCase() == moves[compMove]) {
+    console.log("Comp Move:" + moves[compMove]);
+    console.log("Tie");
+    numTotalGames = numTotalGames + 1;
+    var tieOutput =
+      "It is a tie. <br><br>" +
+      numWins +
+      " of " +
+      numTotalGames +
+      " Games Won<BR>" +
+      "<br>" +
+      "Now you can type scissors, paper or stone to play another round!";
+    return tieOutput;
+  }
+
+  //If User wins
+  if (
+    (input.toLowerCase() == "stone" && moves[compMove] == "scissors") ||
+    (input.toLowerCase() == "scissors" && moves[compMove] == "paper") ||
+    (input.toLowerCase() == "paper" && moves[compMove] == "stone")
```

Comment:
as input.toLowerCase() and moves[compMove] are used repeatedly. they can be made into their own variables above line 36 for better readability and abstraction